### PR TITLE
Create strings.xml files for project name translations

### PR DESCRIPTION
**Pull request by amanj120**
When a Godot game is being exported, one of the processes that takes place is that the resource table is modified to have different translations of the project name. This PR accomplishes the same purpose by writing strings.xml files in the Gradle project directory.

Note: This PR builds off of #39864, the `store_string_at_path` method will be provided through that PR. 

For more details on the final scope of this project, read [this comment](https://github.com/godotengine/godot-proposals/issues/342#issuecomment-648390053).

The `_create_project_name_strings_files` method isn't currently called anywhere in the code, but it will be useful in subsequent PR's.

We are working on this PR with @m4gr3d, as explained [here](https://github.com/godotengine/godot-proposals/issues/342#issuecomment-643655563)

### CODE REVIEW

File: platform/android/export/export.cpp
Comment (amanj120) [inline platform/android/export/export.cpp:747]: Will be removed once #39864 is merged

File: platform/android/export/export.cpp
Comment (amanj120) [inline platform/android/export/export.cpp:1361]: For standalone methods like this one, would it be cleaner to put them in a separate file, i.e. `gradle_export_util.h`?

**Review by m4gr3d — CHANGES_REQUESTED** [review]
The approach used here has visibility issues. 

Unless a contributor is familiar with the added logic here, she/he may inadvertently create a string resource file in the project and be confused when the end result is not as expected since this logic would overwrite that file.

Even when a contributor is familiar with this logic, the addition of new string values now becomes a cumbersome task.

What about the approach of having the string  resources already in the gradle project (e.g: moving them from the `lib` to the `app` module) and just overwriting the target string value?

**Comment by amanj120** [comment]
If we move the string resources from the `lib` module to the `app` module, then the legacy build system no longer works. The [`_fix_resources`](https://github.com/amanj120/godot/blob/3.2/platform/android/export/export.cpp#L1622) method relies on the resource table being pre-populated with values, which only happens if the `strings.xml` files exist inside the `lib` module.

The workaround to this issue is that modifications to `strings.xml` files only be made after the Gradle project has been setup by the Godot Editor.

File: core/io/xml_parser.cpp
Comment (akien-mga) [inline core/io/xml_parser.cpp:56]: We already have `String::xml_escape()` for this purpose.

File: platform/android/export/export.cpp
Comment (akien-mga) [inline platform/android/export/export.cpp:748]: Function parameters should be prefixed with `p_` (or `r_` if they are "return" parameters that get mutated in the function).

```suggestion
	static Error store_string_at_path(const String &p_path, const String &p_data, int p_compression_method = Z_DEFLATED) {
```

File: platform/android/export/export.cpp
Comment (akien-mga) [inline platform/android/export/export.cpp:1367]: What's the reference for this list? Is it a fixed list of locales supported on Android?

It's surprising to see `es` + `es_ES`, but no distinction for `pt_BR` and `pt_PT`.

File: platform/android/export/export.cpp
Comment (akien-mga) [inline platform/android/export/export.cpp:1389]: Nitpick, but for comments we prefer having a space after `//`, unless it's actual code being commented out:
```suggestion
				// TODO: Once the legacy build system is removed, get rid of this else branch.
```

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp:1367]: This list of locales is taken directly from the strings.xml files available in [this directory](https://github.com/amanj120/godot/tree/3.2/platform/android/java/lib/res)

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp:748]: Thank you for the clarification, I will follow this from now on.

File: platform/android/export/export.cpp
Comment (amanj120) [inline platform/android/export/export.cpp:1383]: @m4gr3d I followed your suggestion and moved the `strings.xml` files to the `app` module. This is the new solution, can you take a look?

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp:1367]: If there are more locales you think are necessary, please let me know, and I can add them.

**Comment by amanj120** [comment]
> The approach used here has visibility issues.
> 
> Unless a contributor is familiar with the added logic here, she/he may inadvertently create a string resource file in the project and be confused when the end result is not as expected since this logic would overwrite that file.
> 
> Even when a contributor is familiar with this logic, the addition of new string values now becomes a cumbersome task.
> 
> What about the approach of having the string resources already in the gradle project (e.g: moving them from the `lib` to the `app` module) and just overwriting the target string value?

I'm replying to this to indicate that we've resolved this issue.

File: platform/android/export/export.cpp
Comment (akien-mga) [reply platform/android/export/export.cpp:1366]: Well is there any Android reference that describes what locales should be handled?

File: platform/android/export/export.cpp
Comment (m4gr3d) [reply platform/android/export/export.cpp:1361]: Sounds like a good idea!

File: platform/android/export/export.cpp
Comment (m4gr3d) [reply platform/android/export/export.cpp:1366]: Also since this list doesn't change, it should be a constant outside of the method.

File: platform/android/export/export.cpp
Comment (m4gr3d) [inline platform/android/export/export.cpp:1373]: This string should be a constant as well.

File: platform/android/export/export.cpp
Comment (m4gr3d) [inline platform/android/export/export.cpp:2356]: This should be behind a flag, and you're missing the follow-up logic to disable `_fix_resources` for the app name if this is enabled.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp:1366]: There are a lot of differing sources on the full locale list for Android, none of them seem to be fully encompassing. This is a list from the [Google Play Console Help Page](https://support.google.com/googleplay/android-developer/table/4419860?hl=en). This is a [stackoverflow page](https://stackoverflow.com/questions/7973023/what-is-the-list-of-supported-languages-locales-on-android) about the same topic, but it has many more locales listed.

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp:2362]: This line of code is inside of a block that's behind a `use_custom_build` flag. I pushed another commit just now that disables the future call to `_fix_resources` if `use_custom_build` is true.

File: platform/android/export/export.cpp
Comment (amanj120) [reply platform/android/export/export.cpp]: I changed the logic here a little bit. Instead of having a hardcoded list of locales, what happens now is that this function looks inside the `res://android/build/res` directory for all directories that start with `"values-"`. This means that if at any point in the future, Godot decides to support additional locales, the only change that needs to be made is adding the `platform/android/java/app/res/values-<new locale>/godot_project_name_string.xml` file for the new locale.

File: platform/android/export/export.cpp
Comment (thanwah) [inline platform/android/export/export.cpp:1377]: To reduce nesting maybe and also log in this case?

```
if (!da) {
  return;
}
```

File: platform/android/export/export.cpp
Comment (thanwah) [inline platform/android/export/export.cpp:1384]: To reduce nesting:

```
if (!file.begins_with("values-")) {
  break;
}
```

Comment (amanj120) [reply platform/android/export/export.cpp:1384]: I don't think this achieves the same behavior. For example, the `res/` directory will have directories such as `mipmap`. However, those `mipmap` directories may be given to us by `da->get_next()` before we are done going through all the `values-xx` files. In that scenario, we would not want to break out of the loop.

Comment (thanwah) [reply platform/android/export/export.cpp:1384]: Oops, my bad - I had meant to suggest `continue;` here.

Comment (amanj120) [reply platform/android/export/export.cpp:1384]: But then the loop will never end.

File: platform/android/export/export.cpp
Comment (thanwah) [inline platform/android/export/export.cpp:1387]: I forgot to mention that there are a couple other cases besides locales that may be in a "values-" prefixed folder, e.g. "values-v29" for SDK 29+ resources. Currently the only contents of [Godot's res directory](https://github.com/godotengine/godot/tree/master/platform/android/java/lib/res) are translated strings, but to make this a bit more future proof, could check that `locale` matches a "[a-z]{2}(-[A-Z]{2})?" type pattern. (I didn't test that regex.)

Alternatively, could just add a note/TODO saying that the code assumes all "values-" folders are locales.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

**Comment by amanj120** [comment]
NOTE: Do not merge this PR until #40167 is backported to the 3.2 branch. Look at #40260 for what this PR should look like once #40167 is backported.

**Comment by akien-mga** [comment]
Changes look good overall, though I have the same nitpicks on some of the `xml` files as in #40260.

**Comment by amanj120** [comment]
By merging the `3.2-android-app-bundle` branch into my `bundle_pr_resources` branch, I think I caused a bunch of different commits to be included in this PR that are unrelated to the scope of this PR. The reason I think this is because I am unable to squash the three different commits in this branch into one single commit. 

If this turns out to be an issue, I've created another [`bundle_pr_resources_v2`](https://github.com/godotengine/godot/compare/3.2-android-app-bundle...amanj120:bundle_pr_resources_v2) branch that doesn't have these issues, and is word for word the same as this branch. I can quickly create a separate PR for that branch and close this one.

**Review by m4gr3d — APPROVED** [review]
Can you squash the commits for this PR.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
